#### Практическое задание: Урок 3. Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных


##### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [1]:
import numpy as np
import pandas as pd

In [2]:
Salary = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

In [3]:
mean_salary = Salary.sum()/len(Salary)
mean_salary

65.3

##### Cреднее арифметическое выборки выпускников равно 65.3

Найдем среднее квадратичное отклонение для выборки (S):

In [4]:
A = ((Salary - Salary.mean())**2).sum()
print(A)
S2 = A / (len(Salary))
print(S2)
S = np.sqrt(S2)
print(S)

19002.2
950.11
30.823854398825596


##### Cреднее квадратичное отклонение для выборки (S) равно 30.823854398825596

Смещенная оценка дисперсии равна среднему квадратичному отклонению, возведенному в квадрат:

In [10]:
Var = S2
Var

950.11

Для нахождения НЕсмещенной оценки дисперсии делим сумму квадратов разностей всех величин со средним значением не на  n , а на  n−1:

In [11]:
Var2 = A / (len(Salary) - 1)
Var2

1000.1157894736842

##### Смещенная оценка дисперсии равна 950.11, НЕсмещенная - 1000.1157894736842

##### 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [12]:
from math import factorial

In [13]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Возможные варианты 3 белых мячей из выбранных 6 мячей, вытащенных из 2 ящиков:

Вариант №1: 1 ящик - 2 белых, 2 ящик - 1 белый

In [14]:
P11 = combinations(5, 2)/ combinations(8, 2)
print(P11)
# для 2 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 1 белый 2. вытащили 3 НЕбелых
P12 = combinations(5, 1) * combinations(7, 3)/combinations(12, 4)
print(P12)
P1 = P11*P12
print(P1)

0.35714285714285715
0.35353535353535354
0.12626262626262627


Вариант №2: 1 ящик - 0 белых, 2 ящик - 3 белых

In [15]:
P21 = combinations(3, 2)/ combinations(8, 2)
print(P21)
# для 2 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 3 белых 2. вытащили 1 НЕбелый
P22 = combinations(5, 3) * combinations(7, 1)/combinations(12, 4)
print(P22)
P2 = P21*P22
print(P2)

0.10714285714285714
0.1414141414141414
0.01515151515151515


Вариант №3: 1 ящик - 1 белый, 2 ящик - 2 белых

In [16]:
# для 1 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 1 белый 2. вытащили 1 НЕбелый
P31 = combinations(3, 1)* combinations(5, 1)/combinations(8, 2)
print(P31)
# для 2 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 2 белых 2. вытащили 2 НЕбелых
P32 = combinations(5, 2) * combinations(7, 2)/combinations(12, 4)
print(P32)
P3 = P31*P32
print(P3)

0.5357142857142857
0.42424242424242425
0.22727272727272727


Считаем совокупную вероятность 3 вариантов событий:

In [17]:
P = P1+P2+P3
P

0.3686868686868687

##### Ответ: вероятность того, что 3 мяча белые равна 0.3686868686868687

##### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Вероятность выстрела каждого из трех спортсменов равна 1/3:

In [23]:
pB1 = 1/3
pB2 = 1/3
pB3 = 1/3

Вероятность попадания спортсменов, при условии, что выстрел произведен:

In [24]:
pA_B1 = 0.9
pA_B2 = 0.8
pA_B3 = 0.6

Следовательно, вероятность того, что удачный выстрел произведен (pA):

In [25]:
pA = 1/3*0.9+1/3*0.8+1/3*0.6
pA

0.7666666666666666

Применим формулу Байеса:

In [29]:
pB1A = pB1*pA_B1/pA
pB2A = pB2*pA_B2/pA
pB3A = pB3*pA_B3/pA

In [37]:
print('Ответ:\n')
print('вероятность того, что выстрел произведен: a). первым спортсменом равна', pB1A)
print('вероятность того, что выстрел произведен: б). вторым спортсменом равна', pB2A)
print('вероятность того, что выстрел произведен: в). третьим спортсменом равна', pB3A)

Ответ:

вероятность того, что выстрел произведен: a). первым спортсменом равна 0.391304347826087
вероятность того, что выстрел произведен: б). вторым спортсменом равна 0.3478260869565218
вероятность того, что выстрел произведен: в). третьим спортсменом равна 0.2608695652173913


#### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

In [39]:
# вероятность того, что сдавший студент с факультета:
pB1 = 1/4 # факультет А
pB2 = 1/4 # факультет В
pB3 = 2/4 # факультет С

In [40]:
# вероятность того, что студент сдаст сессию, при условии что он с факультета:
pA_B1 = 0.8 # факультет А
pA_B2 = 0.7 # факультет B
pA_B3 = 0.9 # факультет C

In [41]:
# вероятность того, что студент сдал сессию (pA):
pA = 1/4*0.8+1/4*0.7+2/4*0.9
pA

0.825

In [42]:
# Применим формулу Байеса:
pB1A = pB1*pA_B1/pA
pB2A = pB2*pA_B2/pA
pB3A = pB3*pA_B3/pA

In [43]:
print('Ответ:\n')
print('вероятность того, что студент, сдавший сессию, учится: a). на факультете A равна', pB1A)
print('вероятность того, что студент, сдавший сессию, учится: a). на факультете B равна', pB2A)
print('вероятность того, что студент, сдавший сессию, учится: a). на факультете B равна', pB3A)

Ответ:

вероятность того, что студент, сдавший сессию, учится: a). на факультете A равна 0.24242424242424246
вероятность того, что студент, сдавший сессию, учится: a). на факультете B равна 0.21212121212121213
вероятность того, что студент, сдавший сессию, учится: a). на факультете B равна 0.5454545454545455


##### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

a) совместная вероятность 3 событий - все детали выйдут из строя:

In [21]:
P1 = 0.1*0.2*0.25
P1

0.005000000000000001

б) только две детали

In [2]:
# вариант 1: вышли из строя 1 и 2 деталь
p1 = 0.1*0.2*0.75
# вариант 2: вышли из строя 1 и 3 деталь
p2 = 0.1*0.8*0.25
# вариант 3: вышли из строя 2 и 3 деталь
p3 = 0.9*0.2*0.25
# совместная вероятность 3х возможных вариантов:
P2=p1+p2+p3
P2

0.08000000000000002

в) хотя бы одна деталь

In [23]:
# посчитаем вероятность, что ни одна деталь не выйдет из строя:
p = 0.9*0.8*0.75
# хотя бы одна деталь - обратная величина p:
P3 = 1 - p
P3

0.45999999999999996

г) от одной до двух деталей

In [3]:
# это совместная вероятность выхода из строя 1 детали и 2 деталей. Посчитаем вероятность выхода из строя 1 детали:
p = 0.1*0.8*0.75 + 0.9*0.2*0.75 + 0.9*0.8*0.25
# вероятность выхода из строя 2х деталей мы посчитали выше (Р2). Считаем совместную вероятность 2х событий:
P4 = p+P2
P4

0.455